packages/functions/src/dedup.ts

```diff
@@ -73,105 +73,111 @@ export function dedup(_options: DedupOptions = DEDUP_DEFAULTS): Transform {
 function dedupAccessors(document: Document): void {
 	const logger = document.getLogger();
 
-	// Find all accessors used for mesh data.
-	const indicesAccessors: Set<Accessor> = new Set();
-	const attributeAccessors: Set<Accessor> = new Set();
-	const inputAccessors: Set<Accessor> = new Set();
-	const outputAccessors: Set<Accessor> = new Set();
+	// Find all accessors used for mesh and animation data.
+	const indicesMap = new Map<string, Set<Accessor>>();
+	const attributeMap = new Map<string, Set<Accessor>>();
+	const inputMap = new Map<string, Set<Accessor>>();
+	const outputMap = new Map<string, Set<Accessor>>();
 
 	const meshes = document.getRoot().listMeshes();
 	meshes.forEach((mesh) => {
 		mesh.listPrimitives().forEach((primitive) => {
-			primitive.listAttributes().forEach((accessor) => attributeAccessors.add(accessor));
-			const indices = primitive.getIndices();
-			if (indices) indicesAccessors.add(indices);
+			primitive.listAttributes().forEach((accessor) => hashAccessor(accessor, attributeMap));
+			hashAccessor(primitive.getIndices(), indicesMap);
 		});
 	});
 
 	for (const animation of document.getRoot().listAnimations()) {
 		for (const sampler of animation.listSamplers()) {
-			const input = sampler.getInput();
-			const output = sampler.getOutput();
-			if (input) inputAccessors.add(input);
-			if (output) outputAccessors.add(output);
+			hashAccessor(sampler.getInput(), inputMap);
+			hashAccessor(sampler.getOutput(), outputMap);
 		}
 	}
 
-	// Find duplicate accessors of a given type.
-	function detectDuplicates(accessors: Accessor[]): Map<Accessor, Accessor> {
-		const duplicateAccessors: Map<Accessor, Accessor> = new Map();
+	// Add accessor to the appropriate hash group. Hashes are _non-unique_,
+	// intended to quickly compare everything accept the underlying array.
+	function hashAccessor(accessor: Accessor | null, group: Map<string, Set<Accessor>>): void {
+		if (!accessor) return;
+
+		const hash = [
+			accessor.getCount(),
+			accessor.getType(),
+			accessor.getComponentType(),
+			accessor.getNormalized(),
+			accessor.getSparse(),
+		].join(':');
```

Comment:
Since the `join` here converts all to string, it might worth to encode numbers in hex or base64 to have shorter keys for memory effciency.

Reply:
It might, but I suspect it will be hard to accurately measure the time/memory tradeoff with JavaScript tooling here. We can always improve this further in other PRs, but I think I'd prefer to keep this simple for now.